client/src/pages/userpage/UserPage.jsx

```diff
@@ -0,0 +1,205 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import { useOutletContext, Navigate } from "react-router-dom";
+import { useForm } from "react-hook-form";
+import axios from "axios";
+
+export default function UserPage() {
+  const user = useOutletContext();
+  const { currentUser, setCurrentUser } = user;
+  const express = import.meta.env.VITE_API_URL;
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    watch,
+  } = useForm();
+  const requiredFieldError = "This field is required !";
+
+  const onSubmit = async (data) => {
+    const updatedUser = { ...data, role_id: 1, id: currentUser.id };
+
+    try {
+      axios
+        .put(`${express}/api/users/${currentUser.id}`, updatedUser)
+        .finally(() => {
```

Comment:
Using finally is dangerous here, it can generate serious troubles.

.then -> doing stuff if request is ok
.catch -> doing stuff if request is not positive
.finally -> doing stuff even if request is positive OR negative

in your example it will try to set the user even if the request have failed.

Reply:
fixed